arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Result.kt

```diff
@@ -0,0 +1,211 @@
+    unitResult,
+    unitResult
+  ) { a, b, c, _, _, _, _, _, _, _ -> map(a, b, c) }
+
+public inline fun <A, B, C, D, E> Result<A>.zip(
+  b: Result<B>,
+  c: Result<C>,
+  d: Result<D>,
+  map: (A, B, C, D) -> E
+): Result<E> =
+  zip(
+    b,
+    c,
+    d,
+    unitResult,
+    unitResult,
+    unitResult,
+    unitResult,
+    unitResult,
+    unitResult
+  ) { a, b, c, d, _, _, _, _, _, _ -> map(a, b, c, d) }
+
+public inline fun <A, B, C, D, E, F> Result<A>.zip(
+  b: Result<B>,
+  c: Result<C>,
+  d: Result<D>,
+  e: Result<E>,
+  map: (A, B, C, D, E) -> F
+): Result<F> =
+  zip(b, c, d, e, unitResult, unitResult, unitResult, unitResult, unitResult) { a, b, c, d, e, f, _, _, _, _ ->
+    map(
+      a,
+      b,
+      c,
+      d,
+      e
+    )
+  }
+
+public inline fun <A, B, C, D, E, F, G> Result<A>.zip(
+  b: Result<B>,
+  c: Result<C>,
+  d: Result<D>,
+  e: Result<E>,
+  f: Result<F>,
+  map: (A, B, C, D, E, F) -> G
+): Result<G> =
+  zip(b, c, d, e, f, unitResult, unitResult, unitResult, unitResult) { a, b, c, d, e, f, _, _, _, _ ->
+    map(
+      a,
+      b,
+      c,
+      d,
+      e,
+      f
+    )
+  }
+
+public inline fun <A, B, C, D, E, F, G, H> Result<A>.zip(
+  b: Result<B>,
+  c: Result<C>,
+  d: Result<D>,
+  e: Result<E>,
+  f: Result<F>,
+  g: Result<G>,
+  map: (A, B, C, D, E, F, G) -> H
+): Result<H> =
+  zip(b, c, d, e, f, g, unitResult, unitResult, unitResult) { a, b, c, d, e, f, g, _, _, _ -> map(a, b, c, d, e, f, g) }
+
+public inline fun <A, B, C, D, E, F, G, H, I> Result<A>.zip(
+  b: Result<B>,
+  c: Result<C>,
+  d: Result<D>,
+  e: Result<E>,
+  f: Result<F>,
+  g: Result<G>,
+  h: Result<H>,
+  map: (A, B, C, D, E, F, G, H) -> I
+): Result<I> =
+  zip(b, c, d, e, f, g, h, unitResult, unitResult) { a, b, c, d, e, f, g, h, _, _ -> map(a, b, c, d, e, f, g, h) }
+
+public inline fun <A, B, C, D, E, F, G, H, I, J> Result<A>.zip(
+  b: Result<B>,
+  c: Result<C>,
+  d: Result<D>,
+  e: Result<E>,
+  f: Result<F>,
+  g: Result<G>,
+  h: Result<H>,
+  i: Result<I>,
+  map: (A, B, C, D, E, F, G, H, I) -> J
+): Result<J> =
+  zip(b, c, d, e, f, g, h, i, unitResult) { a, b, c, d, e, f, g, h, i, _ -> map(a, b, c, d, e, f, g, h, i) }
+
+public inline fun <A, B, C, D, E, F, G, H, I, J, K> Result<A>.zip(
+  b: Result<B>,
+  c: Result<C>,
+  d: Result<D>,
+  e: Result<E>,
+  f: Result<F>,
+  g: Result<G>,
+  h: Result<H>,
+  i: Result<I>,
+  j: Result<J>,
+  map: (A, B, C, D, E, F, G, H, I, J) -> K
+): Result<K> = Nullable.zip(
+  getOrNull(),
+  b.getOrNull(),
+  c.getOrNull(),
+  d.getOrNull(),
+  e.getOrNull(),
+  f.getOrNull(),
+  g.getOrNull(),
+  h.getOrNull(),
+  i.getOrNull(),
+  j.getOrNull(),
+  map
+)?.let { success(it) } ?: composeErrors(
+  exceptionOrNull(),
+  b.exceptionOrNull(),
+  c.exceptionOrNull(),
+  d.exceptionOrNull(),
+  e.exceptionOrNull(),
+  f.exceptionOrNull(),
+  g.exceptionOrNull(),
+  h.exceptionOrNull(),
+  i.exceptionOrNull(),
+  j.exceptionOrNull(),
+)!!.let(::failure)
```

Comment:
I had a random thought, not blocking of course. I was wondering if it would make it easier to read these zip implementations if we use computations instead of delegate to higher arity functions,

Reply:
I choose the most effiecient implementation that was still somewhat readablee and short here. Writing this as a `nullable { }` will suffer from the same problem. The only way to write this properly is to be fully exhaustive with `fold` but that exploses quickly.